File: 1098-largest-unique-number/1098-largest-unique-number.java
class Solution {
    public int largestUniqueNumber(int[] nums) {
        short[] count = new short[1001];

        for (int x : nums) {
            count[x]++;
        }

        for (int i = 1000 ; i >= 0; i --) {
            if(count[i] == 1) {
                return i;
            }
        }

        return -1;
    }
}

File: 1098-largest-unique-number/1098-largest-unique-number.cpp
class Solution {
public:
    int largestUniqueNumber(vector<int>& nums) {
        int n = nums.size();

        unordered_map<int,int> mp;

        for(int i = 0; i < n; i++){
            mp[nums[i]]++;
        }

        int ans = -1;
        for(auto x : mp){
            if(x.second == 1){
                ans = max(x.first, ans);
            }
        }

        return ans;
    }
};